pkg/routeagent_driver/handlers/healthchecker/healthchecker_test.go

```diff
@@ -0,0 +1,409 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package healthchecker_test
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	submarinerv1 "github.com/submariner-io/submariner/pkg/apis/submariner.io/v1"
+	fakeClient "github.com/submariner-io/submariner/pkg/client/clientset/versioned/fake"
+	submarinerv1client "github.com/submariner-io/submariner/pkg/client/clientset/versioned/typed/submariner.io/v1"
+	"github.com/submariner-io/submariner/pkg/event"
+	eventtesting "github.com/submariner-io/submariner/pkg/event/testing"
+	"github.com/submariner-io/submariner/pkg/pinger"
+	"github.com/submariner-io/submariner/pkg/pinger/fake"
+	"github.com/submariner-io/submariner/pkg/routeagent_driver/handlers/healthchecker"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
+	dynamicfake "k8s.io/client-go/dynamic/fake"
+	kubeScheme "k8s.io/client-go/kubernetes/scheme"
+)
+
+const (
+	namespace       = "submariner"
+	remoteClusterID = "west"
+	healthCheckIP1  = "1.1.1.1"
+	healthCheckIP2  = "2.2.2.2"
+	localNodeName   = "nodeName"
+)
+
+var _ = Describe("RouteAgent syncing", func() {
+	t := newTestDriver(true)
+	It("should create a RouteAgent resource", func() {
+		var routeAgent *submarinerv1.RouteAgent
+		Eventually(func() bool {
+			var err error
+			routeAgent, err = t.client.Get(context.TODO(), localNodeName, metav1.GetOptions{})
+			return err == nil && routeAgent != nil
```

Comment:
We should fail on unexpected error.

```suggestion
			routeAgent, err = t.client.Get(context.TODO(), localNodeName, metav1.GetOptions{})
			if apierrors.IsNotFound(err) {
			    return false
			}
			
			Expect(err).To(Succeed())
			return true
```

Reply:
Done.